dessia_common/optimization.py

```diff
@@ -124,33 +121,41 @@ def cma_bounds(self):
         return [[0] * len(self.optimization_bounds),
                 [1] * len(self.optimization_bounds)]
 
-    def optimize_gradient(self, method: str = 'L-BFGS-B'):
-        x0 = npy.random.random(self.number_parameters)
+    def optimize_gradient(self, method: str = 'L-BFGS-B', x0: List[float] = None):
+        x0 = self._init_gradient(x0)
         bounds = self.scipy_bounds()
-        result = scipy.optimize.minimize(self.objective_from_dimensionless_vector,
-                                         x0, bounds=bounds, method=method)
 
-        attributes_values = self.vector_to_attributes_values(
-            self.dimensionless_vector_to_vector(result.x))
+        result = scipy.optimize.minimize(self.objective_from_dimensionless_vector, x0, bounds=bounds, method=method)
 
+        attributes_values = self.vector_to_attributes_values(self.dimensionless_vector_to_vector(result.x))
         model = self.instantiate_model(attributes_values)
         return model, result.fun
 
-    def optimize_cma(self):
+    def _init_gradient(self, x0: List[float] = None):
+        if x0 is None:
+            return npy.random.random(self.number_parameters)
```

Comment:
a little bit overkill. why not a generic method random_x ?

Reply:
Merged method _init_gradient into optimize_gradient